tests/components/ruuvi_gateway/test_config_flow.py

```diff
@@ -0,0 +1,103 @@
+"""Test the Ruuvi Gateway config flow."""
+from unittest.mock import patch
+
+from aioruuvigateway.excs import CannotConnect, InvalidAuth
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components import dhcp
+from homeassistant.components.ruuvi_gateway.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .consts import BASE_DATA, EXPECTED_TITLE, GATEWAY_MAC, GET_GATEWAY_HISTORY_DATA
+from .utils import patch_gateway_ok, patch_setup_entry_ok
+
+
+@pytest.mark.parametrize("method", ["user", "dhcp"])
+async def test_ok_setup(hass: HomeAssistant, method: str) -> None:
+    """Test we get the form."""
+    if method == "user":
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+        )
+        data = entry = BASE_DATA
+    elif method == "dhcp":
+        dhcp_ip = "1.2.3.4"
+        data = entry = {**BASE_DATA, "host": dhcp_ip}
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=dhcp.DhcpServiceInfo(
+                hostname="RuuviGateway1234",
+                ip=dhcp_ip,
+                macaddress="12:34:56:78:90:ab",
+            ),
+            context={"source": config_entries.SOURCE_DHCP},
+        )
+    else:
+        raise NotImplementedError("...")
```

Comment:
Please avoid branching like this in tests.  It would be better to do two separate tests instead or parametrize all the args instead.